pkg/pci/address/address.go

```diff
@@ -13,7 +13,7 @@ import (
 
 var (
 	regexAddress *regexp.Regexp = regexp.MustCompile(
-		`^(([0-9a-f]{0,4}):)?([0-9a-f]{2}):([0-9a-f]{2})\.([0-9a-f]{1})$`,
+		`^(([0-9a-f]{0,8}):)?([0-9a-f]{2}):([0-9a-f]{2})\.([0-9a-f]{1})$`,
```

Comment:
@ianb-mp OK, so to be fully compliant with the PCI-X spec, I believe this needs to be:

```
^(1?)(([0-9a-f]{0,4}):)?([0-9a-f]{2}):([0-9a-f]{2})\.([0-9a-f]{1})$
```

Because, according to my research, the domain part of the PCI address is 4 hexadigits *OR* a `1` followed by 4 hexadigits (for the cases where the PCI extension configuration space is used)

Reply:
ok, I've pushed another commit with that change however moved the `1?` inside the existing capture group. I also added a test.